### DISCUSSION TITLE
"Open Nextcloud AIO Interface"-button links to a Non-Existing Page

### DISCUSSION THREAD

**Discussion by JeCheeseSmith** [nextcloud/all-in-one#2802] [Questions] [answered]
Hi!

I succesfully managed to setup Nginx as a reverse proxy for my Nextcloud Instance and I now acces my AIO Interface via: domain.tld:8443. However If I click on the button present in the Administration Section, it redirected me to a page saying "Page not found  The page could not be found on the server."

I found in the README to stop the containers, so I did. Then restarted my mastercontainer and launched the containers again via the newly setupped domain for the AIO interface. However, the button in the Administration page still doesn't work. 

I found this post where it is considered an edge case: https://github.com/nextcloud/all-in-one/discussions/1453

I use the same domain for my nextcloud instance and AIO Interface. It only differs in port :8443 (AIO Instance)

Should I consider this as a bug?

**Comment by Zoey2936** [comment]
can you please show your nginx conf (complete) and your compose file / docker run command?

**Reply by JeCheeseSmith** [reply]
Ofc!

I used the provied docker run command from the Proxy Documentation:

```
# For Linux:
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=0.0.0.0 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

Using NGINX, I setted up 2 basic blocks first:

```
server {
    listen 80;
    listen [::]:80;
    server_name <domain.tld> <www.domain.tld>;
    return 302 https://$server_name$request_uri;
}

server {
    # SSL configuration

    listen 443 ssl http2;
    listen [::]:443 ssl http2;
    ssl_certificate /etc/ssl/cert.pem;
    ssl_certificate_key /etc/ssl/key.pem;

    server_name <domain.tld> <www.domain.tld>;

    root /var/www/<my domain>/html;
    index index.html index.htm index.nginx-debian.html;

    location / {
            try_files $uri $uri/ =404;
    }
}
```

Then 2 blocks from the Proxy Documentation (2nd as suggested in https://github.com/nextcloud/all-in-one/discussions/2801#discussioncomment-6209873):

```
server {
    if ($scheme = "http") {
        return 301 https://$host$request_uri;
    }

    listen 443 ssl http2;      # for nginx versions below v1.25.1
    listen [::]:443 ssl http2; # for nginx versions below v1.25.1 - comment to disable IPv6

    server_name <cloud.domain.tld>;

    location / {
        proxy_pass http://127.0.0.1:11000$request_uri;

        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Port $server_port;
        proxy_set_header X-Forwarded-Scheme $scheme;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Accept-Encoding "";
        proxy_set_header Host $host;

        client_body_buffer_size 512k;
        proxy_read_timeout 86400s;
        client_max_body_size 0;

        # Websocket
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
    }

    ssl_certificate /etc/letsencrypt/live/<filled in my own domain>/fullchain.pem;   # managed by certbot on host machine
    ssl_certificate_key /etc/letsencrypt/live/<filled in my own domain>/privkey.pem; # managed by certbot on host machine

    ssl_session_timeout 1d;
    ssl_session_cache shared:MozSSL:10m; # about 40000 sessions
    ssl_session_tickets off;

    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GC
    ssl_prefer_server_ciphers on;
}

server {
    if ($scheme = "http") {
        return 301 https://$host$request_uri;
    }

    listen 8443 ssl http2;      # for nginx versions below v1.25.1
    listen [::]:8443 ssl http2; # for nginx versions below v1.25.1 - comment to disable IPv6

    server_name <cloud.domain.tld>;

    location / {
        proxy_pass https://127.0.0.1:8080$request_uri;

        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Port $server_port;
        proxy_set_header X-Forwarded-Scheme $scheme;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Accept-Encoding "";
        proxy_set_header Host $host;

        client_body_buffer_size 512k;
        proxy_read_timeout 86400s;
        client_max_body_size 0;

        # Websocket
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
    }

    ssl_certificate /etc/letsencrypt/live/<filled in my own domain>/fullchain.pem;   # managed by certbot on host machine
    ssl_certificate_key /etc/letsencrypt/live/<filled in my own domain>/privkey.pem; # managed by certbot on host machine

    ssl_session_timeout 1d;
    ssl_session_cache shared:MozSSL:10m; # about 40000 sessions
    ssl_session_tickets off;

    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GC
    ssl_prefer_server_ciphers on;
}
```

I think its good to mention I'm using CloudFlare while I took the caveats, mentioned in the README, in mind. 

Please also keep in mind I'm still new to Webservers and might not understand everything out of the box. If there's any info or anything I can do, please let me know!

**Reply by Zoey2936** [reply]
can you please stop the containers over your aio panel on port 8443 and start it again over this url?

**Reply by JeCheeseSmith** [reply]
Yes, I just restarted all the containers using the panel on port 8443, nothing has changed 😅

**Comment by szaimen** [comment]
Hi, can you follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things?

**Reply by JeCheeseSmith** [reply]
Hi!

I followed the debug guide:

- [x] 1: Followed the provided documentation 
- [x] 2: Used the default prescribed port (11000)
- [x] 3: Didn't change the APACHE binding, used 0.0.0.0 as in the docker run command
- [x] 4: I suppose the ports match (see my Nginx Conf files)
- [x]  5: Reverse Proxy (Nginx) runs on the host system
- [ ] 6: I already had a working Nextcloud Instance, I only changed the mastercontainer (created a new one and let it run behind Nginx). Might this cause any issues? (Nextcloud itsself does be working fine) Should I maybe recreate those containers somehow? (Does it then affect my docker volumes?)
- [x] 7: I run Nginx directly on my host, not in a container.
- [x] 8: I didn't knew what CGNAT was, I suppose thats not the case with my VPS (hosted by [Contabo)](https://contabo.com/en) )
- [x]  9: I'm using Cloudflare so I changed my docker run command to:

```
# For Linux:
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=0.0.0.0 \
--env SKIP_DOMAIN_VALIDATION=true \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
``` 
This didn't fix it.

- [x]  10: Ports 80/tcp & 443/tcp are open. 
- [x] 11: I have a IPv6- & IPv4-adress
- [ ] 12: I'd rather not reset everything because there already quit the data in my Nextcloud.

**Reply by szaimen** [reply]
You say it links to a non-existing page? Where does it link to? It should redirect you to the domain that you initially entered in the AIO interface that should be used by Nextcloud. If it should be wrong, you can adjust it by so: https://github.com/nextcloud/all-in-one#how-to-change-the-domain

**Reply by JeCheeseSmith** [reply]
Yes it links me to a non-existing page, I get that as an error from Nextcloud ittself. (See screenshot) 
![image](https://github.com/nextcloud/all-in-one/assets/107920501/17be34c0-871e-4d3c-9e65-9f97fe9d2ce4)

I checked the config and the link to AIO-Interface was: "cloud.domain.tld:" , I changed it to "cloud.domain.tld:8443". (Which I tried before). That didn't work, it changed itsself back to "cloud.octubre.eu:" (without the port 8443)

I also edited it to: "cloud.domain.tld:8443/containers" but it still changes itsself again to "cloud.domain.tld:".

**Reply by szaimen** [reply]
Ah now I see the problem. Can you post the nginx config for the AIO interface here?

**Reply by JeCheeseSmith** [reply]
Yes, I already did as Zoey requested :sweat_smile: https://github.com/nextcloud/all-in-one/discussions/2802#discussioncomment-6211031

**Reply by szaimen** [reply, marked as answer]
Can you check if adjusting the Host header in the 8443 config to ` proxy_set_header Host $host:$server_port;` makes any difference?

**Reply by JeCheeseSmith** [reply]
It does make the difference, it works =D Thanks for helping me out!

**Reply by szaimen** [reply]
@Zoey2936 should we adjust this in the rp docs?

**Reply by Zoey2936** [reply]
no, since the docs are for nextcloud (which needs to run on port 443) and not for aio, also I have the experience that using $host:$server_port can cause problems if applications run on default 443

**Reply by szaimen** [reply]
> no, since the docs are for nextcloud (which needs to run on port 443) and not for aio, also I have the experience that using $host:$server_port can cause problems if applications run on default 443

all right. then lets leave it as is for now.
